server/lib/tuist_web/rate_limit/registry.ex

```diff
@@ -0,0 +1,54 @@
+defmodule TuistWeb.RateLimit.Registry do
+  @moduledoc """
+  Rate limiting for registry endpoints.
+  """
+  import Plug.Conn
+
+  alias Tuist.Accounts.AuthenticatedAccount
+  alias Tuist.Projects.Project
+  alias TuistWeb.Authentication
+  alias TuistWeb.RateLimit.InMemory
+
+  def init(opts), do: opts
+
+  def call(%Plug.Conn{} = conn, _opts) do
+    case hit(conn) do
+      {:allow, _count} ->
+        conn
+
+      {:deny, _limit} ->
+        conn
+        |> put_status(:too_many_requests)
+        |> Phoenix.Controller.json(%{
+          message: "You have made too many requests to the registry. Please try again later."
+        })
+        |> halt()
+    end
+  end
+
+  defp hit(conn) do
+    authenticated? = not is_nil(Authentication.authenticated_subject(conn))
+
+    {key, bucket_size} =
+      if authenticated? do
+        {
+          "registry:auth:#{get_subject_id(conn)}",
+          20_000
+        }
+      else
+        {
+          "registry:unauth:#{TuistWeb.RemoteIp.get(conn)}",
+          1_000
+        }
+      end
+
+    InMemory.hit(key, to_timeout(minute: 1), bucket_size)
```

Comment:
We have another module that uses Redis. I'd lean to using that if it's available falling back to `InMemory`. Otherwise the values will get resetted on deploy.